bindings/azure/blobstorage/blobstorage.go

```diff
@@ -462,3 +463,30 @@ func (a *AzureBlobStorage) isValidDeleteSnapshotsOptionType(accessType azblob.De
 
 	return false
 }
+
+func (a *AzureBlobStorage) sanitizeMetadata(metadata map[string]string) map[string]string {
+	for key, val := range metadata {
+		oldkey := key
+		key = strings.Map(func(r rune) rune {
+			if unicode.IsLetter(r) || unicode.IsDigit(r) {
+				return r
+			}
+			return -1
+		}, key)
+
+		if oldkey != key {
+			delete(metadata, oldkey)
+			metadata[key] = val
+		}
+
+		// Remove all non-ascii characters and replace all colons with underscores
+		metadata[key] = strings.Map(func(r rune) rune {
+			if r > unicode.MaxASCII {
+				return -1
+			}
+			return r
```

Comment:
"Replace colon with underscore" needs to be added.

Reply:
No the comment needs to be updated. I used to have this before, but it isn't needed :)

Reply:
Sure, let's update the `special_value` test as well then.

Reply:
Done.